File: 006.rb
# Sum square difference
# Problem 6
# make solution work for any range of numbers

def sum_of_squares(min,max)
  #range.map { |n| n**2}.reduce(:+)
  sum = 0
  (min..max).each do |num|
    sum += num**2
  end
  return sum
end

def square_of_sums(min,max)
  # (range.reduce(:+))**2
  sum = 0
  (min..max).each do |num|
    sum += num
  end
  return sum**2
end

def difference(min, max)
  square_of_sums(min, max) - sum_of_squares(min,max)
end

File: problem_006/006.js
 // Sum square difference
 // Problem 6
 // make solution work for any range of numbers

function squareDiff(max) {
  var sumSquares = 0;
  var squareSums = 0;
  var sum = 0;
  for (i = 1; i <= max; i++) {
    sumSquares += (i* i);
    sum += i;
  }
  squareSums += (sum * sum);
  return squareSums - sumSquares;
}

console.log(squareDiff(10));
console.log(squareDiff(100));
